pkg/ibmvpc/parser.go

```diff
@@ -1486,71 +1447,100 @@ func getLoadBalancerIPs(vpcConfig *vpcmodel.VPCConfig,
 		if err != nil {
 			return nil, err
 		}
-		// first get name, id, address, publicAddress:
-		var name, id, address, publicAddress string
-		pipIndex, original := subnetsWithPrivateIPs[subnet]
-		if original {
-			// subnet has a private IP, we take it from the config
-			pIP := loadBalancerObj.PrivateIps[pipIndex]
-			name, id, address = *pIP.Name, *pIP.ID, *pIP.Address
-			if hasPublicAddress {
-				publicAddress = *loadBalancerObj.PublicIps[pipIndex].Address
+		subnetBlocks := subnetsBlocks.subnetBlocks(*subnetObj.CRN)
+		privateIPAddressesMessage := make([]string, len(subnetBlocks))
+		// we want a pip for every block of every subnet:
+		for blockIndex, subnetBlock := range subnetBlocks {
+			// first get name, id, address, publicAddress:
+			var name, id, address, publicAddress string
+			blockHasPrivateIP := subnetsPIPsAddresses[subnet] != nil &&
+				subnetsPIPsAddresses[subnet].ContainedIn(subnetBlock)
```

Comment:
Please add explanation as to when a block may not have private IP; perhaps it will also be helpful to recap the distinction between block and subnet

Reply:
done

Reply:
Please remind that a Block is an atomic unit w.r.t. the filters/rules

Reply:
done